module/applications/compendium-browser.mjs

```diff
@@ -301,13 +523,26 @@ export default class CompendiumBrowser extends HandlebarsApplicationMixin(Applic
     }
 
     else if ( partId === "filters" ) {
-      context.additional = Array.from(context.filterDefinitions?.entries() ?? []).reduce((obj, [key, data]) => {
-        obj[key] = foundry.utils.mergeObject(data, {
+      context.additional = Array.from(context.filterDefinitions?.entries() ?? []).reduce((arr, [key, data]) => {
+        // Special case handling for 'Feats' tab in basic mode.
+        if ( (types[0] === "feat") && ((key === "category") || (key === "subtype")) ) return arr;
+
+        let sort = 0;
+        switch ( data.type ) {
```

Comment:
Part of the reason I used `Map` for for the additional filter definitions is so it will retain ordering even after the intersection step, so I wonder if we want to handle ordering more explicitly in the filter definitions to give items more control rather than taking it all over here.